learning-paths/learningPathsModel.js

```diff
@@ -0,0 +1,343 @@
+const db = require('../database/dbConfig')
+
+module.exports = 
+{
+    find,
+    findById,
+    add,
+    updatePathById,
+    deletePathById,
+    joinLearningPath,
+    quitLearningPath,
+    addPathTag,
+    deletePathTag,
+    addPathCourse,
+    removePathCourse,
+    updateCourseOrder,
+    findForUserId,
+    findPathItemsForPath,
+    addPathItem,
+    updatePathItem,
+    deletePathItem,
+    updateContentOrder,
+    findForNotUserId,
+    findForOwner
+}
+
+function find() 
+{
+    return db('paths')
+}
+
+async function findForUserId(userId)
+{
+    let usersPaths = await db('paths as p')
+        .join('users_paths as up', 'up.path_id', '=', 'p.id')
+        .select('p.id', 'p.name', 'p.description', 'p.category')
+        .where({'up.user_id': userId})
+    return usersPaths
+}
+
+async function findForOwner(userId)
+{
+    let ownedPaths = await db('paths as p').where({'p.creator_id': userId})
+    for(let i=0; i<ownedPaths.length; i++)
+    {
+        let tempCourses = await findCoursesForPath(ownedPaths[i].id)
+        tempCourses = tempCourses.map(el => el.id)
+        ownedPaths[i].courseIds = tempCourses
+
+        let tempPI = await findPathItemsForPath(ownedPaths[i].id)
+
+        ownedPaths[i].contentLength = tempPI.length + tempCourses.length
+    }
+
+    return ownedPaths
+}
+
```

Comment:
findfornotuserid:  i still wonder if there is a way to query easier with whereNot?